### PR TITLE
Add Testcase to verify JSON support in NashornJs

### PR DESCRIPTION
**Pull request by malakaganga**

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java [hunk 1]
```diff
@@ -55,6 +64,78 @@ public void testSendingPayloadJson() throws Exception {
...
+                + "\"appointment_date\": " + "\"2017-04-02\"\n" + "}";
+        HttpResponse response = doPost(new URL(getApiInvocationURL("nashornJsHandlingNullJsonObject")), payload,
+                httpHeaders);
+        Assert.assertTrue((response.getData().contains("{}")), "Response does not contain "
+                + "the keyword \"{}\". Response: " + response.getData());
+
+    }
+
+    /**
+     * This method check whether given property contains in the logs.
```
Comment (Manuri) [inline integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: Need a new line after method description

Comment (malakaganga) [reply integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/script/NativeJsonSupportByNashornJsTestCase.java]: Done

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsHandlingNullJsonObject.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		mc.setProperty("json", retrivedJsonObject);
+
+		var jsonproperty = mc.getProperty("json");
+		mc.setPayloadJSON(jsonproperty);
+
+	   ]]></script>
+            <respond/>
+        </inSequence>
+    </resource>
+</api>
```
Comment (sumedhassk) [inline integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsHandlingNullJsonObject.xml]: Better to have a new line

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsParse.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		mc.setPayloadJSON(doctor);
+
+	   ]]></script>
+            <log level="custom">
+                <property expression="get-property('json')" name="JSON_TEXT"/>
+            </log>
+            <respond/>
+        </inSequence>
+    </resource>
+</api>
```
Comment (sumedhassk) [inline integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsParse.xml]: Better to have a new line

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsStringify.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		mc.setProperty("json", rettrivedJsonText);
+
+	   ]]></script>
+            <log level="custom">
+                <property expression="get-property('json')" name="JSON_TEXT"/>
+            </log>
+            <respond/>
+        </inSequence>
+    </resource>
+</api>
```
Comment (sumedhassk) [inline integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsStringify.xml]: Better to have a new line

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsHandlingNullJsonObject.xml (hunk 2 above)
Comment (malakaganga) [reply integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsHandlingNullJsonObject.xml]: Hi Sumedha,
Fixed with following PR #1511

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsParse.xml (hunk 3 above)
Comment (malakaganga) [reply integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsParse.xml]: Hi Sumedha,
Fixed with following PR #1511

File: integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsStringify.xml (hunk 4 above)
Comment (malakaganga) [reply integration/mediation-tests/tests-mediator-2/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/api/nashornJsStringify.xml]: Hi Sumedha,
Fixed with following PR #1511

**Pull request merged**